pyccel/codegen/printing/ccode.py

```diff
@@ -981,6 +988,14 @@ def _print_Assign(self, expr):
             return '{};'.format(self._print(expr.rhs))
         lhs = self._print(expr.lhs)
         rhs = expr.rhs
+        if isinstance(rhs, IndexedElement) and isinstance(expr.lhs, Variable):
+            free_code = ''
+            if any(isinstance(i, Slice) for i in rhs.args):
+                free_code += 'free_array(%s);\n' % self._print(expr.lhs.name)
```

Comment:
It would be better to make this:
```
free_code += self._print(Deallocate(expr.lhs))
```
In case we ever want to change `_print_Deallocate`